src/connectors/argentMobile/modal/starknet/adapter.ts

```diff
@@ -102,10 +107,60 @@ export class StarknetAdapter
   }
 
   // StarknetWindowObject
+  request: RequestFn = async (call): Promise<any> => {
+    if (!this.session) {
+      throw new Error("No session")
+    }
+
+    if (call.type === "wallet_requestChainId") {
+      return this.chainId === constants.NetworkName.SN_GOERLI
+        ? constants.StarknetChainId.SN_GOERLI
+        : constants.StarknetChainId.SN_MAIN
+    }
+
+    if (call.type === "wallet_requestAccounts") {
+      return this.accounts
+    }
+
+    if (call.type === "wallet_getPermissions") {
+      if (await this.isPreauthorized()) {
+        return ["accounts"]
+      }
+
+      return []
+    }
+
+    if (call.type === "starknet_addInvokeTransaction") {
+      const { calls } = call.params as AddInvokeTransactionParameters
+
+      return await this.requestWallet({
+        method: "starknet_requestAddInvokeTransaction",
+        params: {
+          accountAddress: this.account.address,
+          executionRequest: {
+            calls: calls?.map(({ contract_address, ...rest }) => ({
+              ...rest,
+              contractAddress: contract_address,
+            })),
+          },
+        },
+      })
+    }
+
+    if (call.type === "starknet_signTypedData") {
+      const params = {
+        accountAddress: this.account.address,
+        typedData: { ...(call.params as TypedData) },
+      }
+
+      const response = (await this.requestWallet({
+        method: call.type,
+        params,
+      })) as { signature: string[] } | string[]
+
+      return "signature" in response ? response.signature : response
+    }
```

Comment:
@bluecco are you also going to add support for `"starknet_supportedSpecs"` call type? The call has no params and the response is an array of strings, eg `["0.6.0"]`

Reply:
will add it

Reply:
@mluisbrown `starknet_supportedSpecs` is already in the mobile app?

getting ```Error: Missing or invalid. request() method: starknet_supportedSpecs``` when sending the request

Reply:
No, I only just made the [PR](https://github.com/argentlabs/ios/pull/5179) for it 😄 But I want to be able to test it...

Reply:
ah ok :D
will push the change ;)